src/server/detail/save_stages_controller.cc

```diff
@@ -169,6 +183,25 @@ GenericError SaveStagesController::Save() {
   return *shared_err_;
 }
 
+size_t SaveStagesController::GetSaveBuffersSize() {
+  std::atomic<size_t> total_bytes{0};
+  if (use_dfs_format_) {
+    auto cb = [this, &total_bytes](ShardId sid) {
+      total_bytes.fetch_add(snapshots_[sid].first->GetSaveBuffersSize(), memory_order_relaxed);
+    };
+    shard_set->RunBriefInParallel([&](EngineShard* es) { cb(es->shard_id()); });
+
+  } else {
+    // When rdb format save is running, there is only one rdb saver instance, it is running on the
+    // connection thread that runs the save command. How can we make sure that the value returned
+    // from GetSaveBuffersSize is valid? as it does not runs on the same thread, but pulling the
+    // pulled bytes values updated by the rdb saver, should we make this value atomic? is there
```

Comment:
@dranikpg Hi, what do you think about this?

Reply:
Making stats.pulled_bytes atomic would be the simplest solution, otherwise we can just store the thread from which save was launched and then dispatch onto that